filters/imagefilter.go

```diff
@@ -48,25 +49,12 @@ func HandleImageResponse(ctx filters.FilterContext, f ImageFilter) {
 
 	rsp.Header.Del("Content-Length")
 
-	in := rsp.Body
-	r, w := io.Pipe()
-	rsp.Body = r
-
-	go handleImageTransform(w, in, f)
+	handleImageTransform(rsp, f)
 }
 
-func handleImageTransform(out *io.PipeWriter, in io.ReadCloser, f ImageFilter) error {
+func handleImageTransform(rsp *http.Response, f ImageFilter) error {
 	var err error
-
-	defer func() {
```

Comment:
we have removed in, out streams here. But directly read from memory (byte []). As rsp.Body is of type 'ReadCloser', to satisfy the type we wrapped the byte slice into a 'ReadCloser' using helper functions provided by iotutil. So I think its safe to remove the code but lets wait till our expert confirm it :-)